src/core/LiveOrderbook.ts

```diff
@@ -170,24 +171,37 @@ export class LiveOrderbook extends Duplex implements Orderbook {
         }
         switch (msg.type) {
             case 'ticker':
-                this.updateTicker(msg as TickerMessage);
+                this.updateTicker(msg);
                 // ticker is emitted in pvs method
                 break;
             case 'snapshot':
-                this.processSnapshot(msg as SnapshotMessage);
+                this.processSnapshot(msg);
                 break;
             case 'level':
-                this.processLevelChange(msg as LevelMessage);
+                this.processLevelChange(msg);
                 this.emit('LiveOrderbook.update', msg);
                 break;
             case 'trade':
                 // Trade messages don't affect the orderbook
                 this.emit('LiveOrderbook.trade', msg);
                 break;
-            default:
-                this.processLevel3Messages(msg as OrderbookMessage);
+            case 'newOrder':
+            case 'orderDone':
+            case 'changedOrder':
+                this.processLevel3Messages(msg);
                 this.emit('LiveOrderbook.update', msg);
                 break;
+            case 'cancelOrder':
+            case 'error':
+            case 'myOrderPlaced':
+            case 'placeOrder':
+            case 'tradeExecuted':
+            case 'tradeFinalized':
+            case 'unknown':
+                break;
+            default:
+                ((_x: never) => { return; })(msg);
```

Comment:
Use the assertNever util?

Reply:
@fb55 Fixed, waiting for the other branch to be merged in. PTAL.